app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx

```diff
@@ -2,12 +2,41 @@ import { Box } from "@mui/material";
 import React from "react";
 import HeaderView from "./HeaderView";
 import { getComponentProps } from "../utils";
+import { usePanelId } from "@fiftyone/spaces";
+import { usePanelEvent } from "@fiftyone/operators";
 
 export default function ImageView(props) {
   const { schema, data } = props;
-  const { height, width, alt } = schema?.view || {};
+  const {
+    height,
+    width,
+    alt,
+    href,
+    operator,
+    prompt = false,
+    params,
+  } = schema?.view || {};
   const imageURI = data ?? schema?.default;
 
+  const panelId = usePanelId();
+  const handleClick = usePanelEvent();
+
+  const onClick = () => {
+    // ordering matters: execute operator first then redirect if defined
+
+    if (operator) {
+      handleClick(panelId, {
```

Comment:
if an operator is provided we should call `handleClick` and provide a callback.

```typescript
    triggerEvent(panelId, {
      operator,
      params,
      prompt: props.prompt, // we could also support this
      callback: (result: OperatorResult) => {
        if (href) window.open(href, "_blank");
      },
    });
```

Reply:
Moved additionally functionality to callback to strictly make sure execution is done in order